### PR TITLE
Symbols: add more APIs

### PR DESCRIPTION
**Pull request by auduchinok**
Adds some APIs to FSharpEntity and FSharpType types that we've been using in our FCS fork.

### CODE REVIEW

File: src/Compiler/Symbols/Symbols.fs [hunk 1]
```diff
@@ -430,18 +430,27 @@ type FSharpEntity(cenv: SymbolEnv, entity: EntityRef) =
...
-        let fail() = invalidOp (sprintf "the type '%s' does not have a qualified name" x.LogicalName)
-#if !NO_TYPEPROVIDERS
-        if entity.IsTypeAbbrev || entity.IsProvidedErasedTycon || entity.IsNamespace then fail()
-        #else
-        if entity.IsTypeAbbrev || entity.IsNamespace then fail()
+
+        let fail () =
+            invalidOp (sprintf $"the type '{x.LogicalName}' does not have a qualified name")
+
+#if !NO_EXTENSIONTYPING
```
Comment (vzarytovskii) [inline src/Compiler/Symbols/Symbols.fs:439]: This was replaced by `NO_TYPEPROVIDERS` some time ago, I believe.

Comment (auduchinok) [reply src/Compiler/Symbols/Symbols.fs]: Updated, thanks! This was cherry-picked from an older branch.

**Comment by auduchinok** [comment]
This is ready.

File: src/Compiler/Symbols/Symbols.fs [hunk 2]
```diff
@@ -2245,6 +2258,11 @@ type FSharpMemberOrFunctionOrValue(cenv, d:FSharpMemberOrValData, item) =
         | V vref -> isForallFunctionTy cenv.g vref.Type
         | _ -> false
 
+    member x.IsRefCell =
```
Comment (dsyme) [inline src/Compiler/Symbols/Symbols.fs:2261]: I think I prefer the name `IsFSharpRefType` for the public API

File: src/Compiler/Symbols/Symbols.fs [hunk 3]
```diff
@@ -2478,6 +2496,21 @@ type FSharpType(cenv, ty:TType) =
         GetSuperTypeOfType cenv.g cenv.amap range0 ty
         |> Option.map (fun ty -> FSharpType(cenv, ty)) 
 
+    member x.StrippedType =
```
Comment (dsyme) [inline src/Compiler/Symbols/Symbols.fs:2499]: Prefer `ErasedType` I think.

File: src/Compiler/Symbols/Symbols.fsi [hunk 4]
```diff
@@ -933,8 +939,8 @@ type FSharpMemberOrFunctionOrValue =
     /// Indicated if this is a value compiled to a method
     member IsValCompiledAsMethod: bool
 
-    /// Indicates if this is a function
-    member IsFunction: bool
```
Comment (dsyme) [inline src/Compiler/Symbols/Symbols.fsi:937]: Why was this removed?

**Review by dsyme — CHANGES_REQUESTED** [review]
Left some comments

File: src/Compiler/Symbols/Symbols.fsi (hunk 4 above)
Comment (dsyme) [reply src/Compiler/Symbols/Symbols.fsi:937]: Strange that tests still passed with this removed....

Comment (auduchinok) [reply src/Compiler/Symbols/Symbols.fsi:937]: For some reason there were two duplicates, and I removed one of them.

Comment (dsyme) [reply src/Compiler/Symbols/Symbols.fsi:937]: Hmmm indicates a bug that some duplicates in signatures are not being detected. Could you add an issue for that please? Thanks

Comment (auduchinok) [reply src/Compiler/Symbols/Symbols.fsi:937]: @dsyme Done.

File: src/Compiler/Symbols/Symbols.fs (hunk 2 above)
Comment (auduchinok) [reply src/Compiler/Symbols/Symbols.fs:2261]: It's possible, yes, but it'd be less convenient. We've been using this API for several years, and I found it useful. Will you accept it with  `IsFSharpRef` name?

Comment (dsyme) [reply src/Compiler/Symbols/Symbols.fs:2261]: Yes, I guess it's OK given the special role the type plays in some corner-case F# language rules

File: src/Compiler/Symbols/Symbols.fsi [hunk 5]
```diff
@@ -1108,6 +1114,12 @@ type FSharpType =
     /// if it is an instantiation of a generic type.
     member BaseType: FSharpType option
 
+    /// Canonical form of the type with abbreviations, measures, and F# tuples and functions erased.
+    member StrippedType: FSharpType
```
Comment (T-Gro) [inline src/Compiler/Symbols/Symbols.fsi:1118]:
```suggestion
    member ErasedType: FSharpType
```

File: tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected [hunk 6]
```diff
@@ -5111,7 +5116,9 @@ FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType Abbreviat
 FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType Instantiate(Microsoft.FSharp.Collections.FSharpList`1[System.Tuple`2[FSharp.Compiler.Symbols.FSharpGenericParameter,FSharp.Compiler.Symbols.FSharpType]])
 FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType Prettify(FSharp.Compiler.Symbols.FSharpType)
 FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType StripAbbreviations()
+FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType StrippedType
```
Comment (T-Gro) [inline tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected:5119]:
```suggestion
FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType ErasedType
```

File: tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected [hunk 7]
```diff
@@ -5111,7 +5116,9 @@ FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType Abbreviat
...
 FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType get_AbbreviatedType()
+FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType get_StrippedType()
```
Comment (T-Gro) [inline tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected:5121]:
```suggestion
FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType get_ErasedType()
```

File: src/Compiler/Symbols/Symbols.fs [hunk 8]
```diff
@@ -2483,6 +2501,21 @@ type FSharpType(cenv, ty:TType) =
...
```
Comment (T-Gro) [inline src/Compiler/Symbols/Symbols.fs:2504]:
```suggestion
    member x.ErasedType=
```

File: tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected (hunk 6 above)
Comment (T-Gro) [inline tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected:5119]:
```suggestion
FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType ErasedType
```

File: tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected (hunk 7 above)
Comment (T-Gro) [inline tests/FSharp.Compiler.Service.Tests/FSharp.CompilerService.SurfaceArea.netstandard.expected:5121]:
```suggestion
FSharp.Compiler.Symbols.FSharpType: FSharp.Compiler.Symbols.FSharpType get_ErasedType()
```

**Comment by auduchinok** [comment]
Hey, sorry for not updating PR earlier, I've had some local issues with FCS solution recently. I can update the needed things if CI is not green after this run

**Review by T-Gro — APPROVED** [review]

**Comment by T-Gro** [comment]
Checks passing, please approve.

**Review by psfinaki — APPROVED** [review]
Thanks Eugene :)

**Comment by psfinaki** [comment]
@dsyme please reapprove

**Comment by dsyme** [comment]
Thanks @psfinaki @T-Gro !
